### PR TITLE
Build repo with clang

### PR DESCRIPTION
**Pull request by mohanson**

### CODE REVIEW

File: Makefile.clang [hunk 1]
```diff
@@ -0,0 +1,238 @@
+CURRENT_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+CC := clang-16
+LD := ld.lld-16
+OBJCOPY := llvm-objcopy-16
+CFLAGS_LLVM := --target=riscv64 -march=rv64imac_zba_zbb_zbc_zbs -Wno-error=unused-command-line-argument
+CFLAGS := ${CFLAGS_LLVM} -fPIC -O3 -nostdinc -nostdlib -nostartfiles -fvisibility=hidden -I deps/ckb-c-stdlib -I deps/ckb-c-stdlib/libc -I deps -I deps/ckb-c-stdlib/molecule -I c -I build -I deps/secp256k1/src -I deps/secp256k1 -g -Wno-nonnull -Wno-unused-function
+LDFLAGS := -Wl,-static -fdata-sections -ffunction-sections -Wl,--gc-sections
```
Comment (XuJiandong) [inline Makefile.clang]: `-fdata-sections -ffunction-sections` are CFLAGS and only `-Wl,--gc-sections` is for linker.

File: Makefile.clang [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+CFLAGS_LIBECC_OPTIMIZED = $(CFLAGS_LIBECC) -DWITH_LL_U256_MONT
+CFLAGS_LINK_TO_LIBECC_OPTIMIZED := ${CFLAGS_LLVM} -fno-builtin -DWORDSIZE=64 -DWITH_CKB -DWITH_BLANK_EXTERNAL_DEPENDENCIES -fno-builtin-printf -I ${LIBECC_OPTIMIZED_PATH}/src -I ${LIBECC_OPTIMIZED_PATH}/src/external_deps
+
+MOLC := moleculec
+MOLC_VERSION := 0.7.0
+
+DOCKER_USER := $(shell id -u):$(shell id -g)
+DOCKER_EXTRA_FLAGS ?=
+# docker pull nervos/ckb-riscv-gnu-toolchain:gnu-bionic-20191012
+BUILDER_DOCKER := nervos/ckb-riscv-gnu-toolchain@sha256:aae8a3f79705f67d505d1f1d5ddc694a4fd537ed1c7e9622420a470d59ba2ec3
```
Comment (XuJiandong) [inline Makefile.clang]: Remove docker.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: Makefile.clang (hunk 1 above)
Comment (mohanson) [reply Makefile.clang]: Here I copied it from: https://github.com/nervosnetwork/ckb-miscellaneous-scripts/blob/master/Makefile#L8

To be honest, I think this is not the only problem in the Original Makefile

File: Makefile.clang (hunk 2 above)
Comment (mohanson) [reply Makefile.clang]: Sure

File: Makefile.clang (hunk 1 above)
Comment (XuJiandong) [reply Makefile.clang]: Fix it if possible. The binary can be larger than expected.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**